### PR TITLE
7 gradient boosting regressor

### LINKED ISSUE #7: Explore Gradient Boosting Regressor
We want to explore Gradient Boosting Regressor for example, [Gradient Boosting, Extreme Gradient Boosting, and Light Gradient Boosting Machine](https://pycaret.readthedocs.io/en/latest/api/regression.html) on hybrid labs data. For that, a python module needed to be called to fit, and predict. The module should:
- [ ] similar to [pycaret](https://pycaret.gitbook.io/docs) to compare different models, see [examples](https://nbviewer.org/github/pycaret/pycaret/blob/master/tutorials/Tutorial%20-%20Regression.ipynb)
- [ ] have a notebook in `docs` folder to show a simple example
- [ ] have tests
- [ ] have an easy interface and run fast (to be discussed)

**Data:**
The data folder `exp699_032024_TUDelft` is available on **Atlas** SharePoint at `Documents > HybridLabs > Example_data`. **NOTE:** The data folder `exp699_032024_TUDelft` is **not** public and cannot be shared with others. The trained model on this data cannot be also shared. For now, no need to store the trained model. 

The data folder `exp699_032024_TUDelft` includes:

- `readme.md`: it contains the experiment details and ML input/output
- `channel.csv:` it contains the variable names and units 
- `exp699.mat:` it contains the data and it is in matlab format.

For reading data in python, see [this notebook](https://github.com/hybridlabs-nl/FOWT-ML/blob/main/docs/notebooks/compare_linear_models.ipynb).

**Literature:**

Some literature is available at on **Atlas** SharePoint at `Documents > HybridLabs > Literature`. The two most related to FOWT are:

- [A hybrid methodology for wind tunnel testing of floating offshore wind turbines](https://doi.org/10.1016/j.oceaneng.2020.107592)
- [A review of modelling techniques for floating offshore wind turbines](https://doi.org/10.1002/we.2701)

### PR DESCRIPTION
**Pull request by vanlankveldthijs**
Resolves #7

Initial experiments for using the gradient boosting regressor.

The sklearn regressor performs similarly to least angle, lasso, or ridge regression, but it requires parameter tuning. It is also slightly slower and requires the MultiOutputRegressor to work on our example dataset.

### CODE REVIEW

**Comment by vanlankveldthijs** [comment]
I also tried to replicate some examples for using the xgboost package. However, I could not yet resolve the issue that it requires feature names.
